include/polygeist/PolygeistOps.td

```diff
@@ -17,10 +17,9 @@ include "mlir/Dialect/LLVMIR/LLVMOpBase.td"
 include "mlir/Dialect/LLVMIR/LLVMOpsInterfaces.td"
 
 def CacheLoad
-    : Polygeist_Op<"cacheload"> {
+    : Polygeist_Op<"cacheload", [NoSideEffect]> {
```

Comment:
This feels too aggressive..?

Reply:
Can the pattern RewritePattern folder etc mess up something if we have NoSideEffects? As far as I can see our code seems fine with it, but I am not sure if something from mlir might be a problem.

We can remove that and just add special cases for cacheload in collectEffects

Reply:
My concern is that any other (or future) part of the code will incorrectly model the semantics, and I'd prefer to see us more explicitly mark where we're making assumptions.